Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift

```diff
@@ -58,7 +62,7 @@ extension Lambda {
         }
 
         /// Reports a result to the Runtime Engine.
-        func reportResults(logger: Logger, invocation: Invocation, result: Result<ByteBuffer?, Error>) -> EventLoopFuture<Void> {
+        func reportResults(logger: Logger, invocation: Invocation, result: Result<ByteBuffer?, Error>, context: BaggageContext) -> EventLoopFuture<Void> {
```

Comment:
in some places we refer to `BaggageContext` as `context` argument and in other places its referred to as `baggage`. I assume we all agree we need to be consistent. Maybe if not too late (and if makes sense) we should just rename the type to the short form `Baggage` and refer to it `baggage` since in most frameworks there will be some kind of `Context` that is the container/carrier. WDYT?

Reply:
ah saw more discussion to this end on the above. looks like this is still a bit of area of confusion so maybe good feedback for the API and naming choices. Lets see if we can come with something that is intuitive otherwise we will end up having this discussion every time

Reply:
Yeah very much agreed... We wrote thought this through and there are documented guidelines here: https://github.com/slashmo/gsoc-swift-baggage-context#argument-namingpositioning

Specifically, it (and any other context) basically always should be context; a parameter should never be called baggage; the ONLY place there can be `baggage` is in a `BaggageContextCarrier` (which are always a "framework context") in order to avoid `context.context` it becomes `context.baggage` but since libraries should accept carriers rather than baggage itself most of the time you can always pass it like this `let context: Lambda.Context (is ...Carrier)` into `get(..., context: context)` since http client would accept `...Carrier` (the protocol, rather than the specific context type).

// I'm very open to finding better names. Will try to brainstorm this some more with feedback from this PR.

Reply:
Using _just_ the name Baggage we should think some more about if that's an option... That's going to be a standard name: https://w3c.github.io/baggage/ and it matches pretty much what we are TBH, a bag of pretty structured data.

BUT there is also https://www.w3.org/TR/trace-context/ and their relationship still remains messy (in the standards to be honest).

BaggageContext is a term known from PivotTracing and TracingPlane and somewhat "known" in the tracing space...

Open question really, but we need to be careful here :~

Reply:
personally, naming it just `Baggage` and the a framework context is a `BaggageCarrier` (cute) or `BaggageContainer` (technical) seems compelling but I may be missing use cases where this does not work?